src/rust/catcollar/futures/accept.rs

```diff
@@ -64,46 +68,63 @@ impl AcceptFuture {
     }
 }
 
-//==============================================================================
+//======================================================================================================================
 // Trait Implementations
-//==============================================================================
+//======================================================================================================================
 
 /// Future Trait Implementation for Accept Operation Descriptors
 impl Future for AcceptFuture {
     type Output = Result<RawFd, Fail>;
 
     /// Polls the underlying accept operation.
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
-        let self_: &AcceptFuture = self.get_mut();
-        match socket::accept(self_.fd as i32) {
+        let self_: &mut AcceptFuture = self.get_mut();
+        match unsafe {
+            let mut address_len: libc::socklen_t = 0;
+            libc::accept(
+                self_.fd,
+                (&mut self_.sockaddr as *mut libc::sockaddr_in) as *mut libc::sockaddr,
+                &mut address_len,
+            )
+        } {
             // Operation completed.
-            Ok(new_fd) => {
+            new_fd if new_fd >= 0 => {
                 trace!("connection accepted ({:?})", new_fd);
 
                 // Set socket options.
                 unsafe {
                     if linux::set_tcp_nodelay(new_fd) != 0 {
-                        warn!("cannot set TCP_NONDELAY option");
+                        let errno: libc::c_int = *libc::__errno_location();
+                        warn!("cannot set TCP_NONDELAY option (errno={:?})", errno);
                     }
                     if linux::set_nonblock(new_fd) != 0 {
-                        warn!("cannot set NONBLOCK option");
+                        let errno: libc::c_int = *libc::__errno_location();
+                        warn!("cannot set O_NONBLOCK option (errno={:?})", errno);
                     }
                     if linux::set_so_reuseport(new_fd) != 0 {
-                        warn!("cannot set SO_REUSEPORT option");
+                        let errno: libc::c_int = *libc::__errno_location();
+                        warn!("cannot set SO_REUSEPORT option (errno={:?})", errno);
                     }
                 }
 
                 Poll::Ready(Ok(new_fd))
             },
-            // Operation in progress.
-            Err(e) if e == Errno::EWOULDBLOCK || e == Errno::EAGAIN => {
-                ctx.waker().wake_by_ref();
-                Poll::Pending
-            },
-            // Operation failed.
-            Err(e) => {
-                warn!("failed to accept connection ({:?})", e);
-                Poll::Ready(Err(Fail::new(e as i32, "operation failed")))
+
+            // Operation not completed, thus parse errno to find out what happened.
```

Comment:
I would put all these comments inside the block to which they apply, as they describe state that is only true inside the block.

Reply:
Sure. I moved them in there.

Reply:
Thanks!